### PR TITLE
Add smartFriendlyCritter, demonstrating continuous pathfinding

### PR DESCRIPTION
**Pull request by AndyTechGuy**
**Requires Terasology/Behaviors#22 !**

## Contains

A new deer character, `smartFriendlyCritter`. This critter follows the character much like `friendlyCritter`, but this critter uses pathfinding to do it, avoiding obstacles and pitfalls in it's path. As well, this critter follows the character from up to 500 blocks away!

This character serves to demonstrate & test the continuous pathfinding abilities of [Terasology/Pathfinding](https://github.com/Terasology/Pathfinding).

### CODE REVIEW

**Review by msteiger — APPROVED** [review]
Looks good to me!

File: src/main/java/org/terasology/minion/move/SetupContinuousTravelNode.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.terasology.logic.behavior.BehaviorAction;
+import org.terasology.logic.behavior.core.Actor;
+import org.terasology.logic.behavior.core.BaseAction;
+import org.terasology.logic.behavior.core.BehaviorState;
+import org.terasology.navgraph.WalkableBlock;
+import org.terasology.pathfinding.model.Path;
+
+/**
+ * Essential setup for a character to travel along a determined path. This is exported from {@link MoveAlongPathNode#construct(Actor)}, which allows for characters to
```
Comment (msteiger) [inline src/main/java/org/terasology/minion/move/SetupContinuousTravelNode.java]: Just a comment: I try to stick to max. 120 chars per line, if possible, so I don't have to scroll horizontally. GitHub shows 130 chars, eclipse uses 120 by default afaik. Especially for block comments it should be easy to do, but it's a matter of personal preference, I guess.

**Review by msteiger — APPROVED** [review]

File: src/main/java/org/terasology/minion/move/SetupContinuousTravelNode.java (hunk 1 above)
Comment (AndyTechGuy) [reply src/main/java/org/terasology/minion/move/SetupContinuousTravelNode.java]: I transferred these nodes to Terasology/Behaviors#22, and I wrapped the comments to less than 120 characters. I've added a line guide to IDEA so all lines will be less than 120 characters in the future 👍

**Comment by Cervator** [comment]
Tested out and merged 👍

**Pull request merged**